src/server/room-controller.js

```diff
@@ -86,6 +73,34 @@ export default class RoomController {
     return res.status(201).end();
   }
 
+  async handleGetRoomPage(req, res) {
+    const { roomId } = req.params;
+    const room = await this.roomService.getRoomById(roomId);
+
+    if (!room) {
+      throw new NotFound();
+    }
+
+    const { _id: userId } = req.user;
+
+    let invitations = null;
+
+    if (room.access === ROOM_ACCESS_LEVEL.private) {
+      const isOwnerOrMember = await this.roomService.isRoomMemberOrOwner(roomId, userId);
+      if (!isOwnerOrMember) {
+        throw new Forbidden();
+      }
+
+      if (room.owner === userId) {
+        invitations = await this.roomService.getRoomInvitations(roomId);
+      }
+    }
+
+    const roomDetails = await this.clientDataMapper.mapRoom(room);
+
+    return this.pageRenderer.sendPage(req, res, PAGE_NAME.room, { roomDetails, invitations });
```

Comment:
Can we just call it `room`? I don't think that name is already taken by a different data structure.

Reply:
Wanted to suggest the same actually.

Reply:
I wanted to keep a distinction between the room without any other data aggregated to it (as it comes from the db) and a room that had more data loaded.
Changed.